File: plugins/adapter.py
# Route Adapters for SpaceDock Backend
# Copyright (c) 2016 The SpaceDock Team
# Licensed under the terms of the MIT License
#
# Name: SpaceDock.extras.adapter
from SpaceDock.formatting import mod_info
from SpaceDock.objects import Mod
from SpaceDock.routing import route

@route('/api/adapter/mods/<modid>')
def mods_adapter(modid):
    """
    Returns information for one mod
    """
    if not modid.isdigit() or not Mod.query.filter(Mod.id == int(modid)).first():
        return {'error': True, 'reasons': ['The modid is invalid']}, 400
    # Get the mod
    mod = Mod.query.filter(Mod.id == int(modid)).first()
    if not mod.published and current_user != mod.user:
        return {'error': True, 'reasons': ['The mod is not published.']}, 400
    return {'error': False, 'count': 1, 'data': mod_info(mod)}

File: plugins/adapter/adapter.go
/*
 Route Adapters for SpaceDock Backend

 SpaceDock-Extras is licensed under the Terms of the MIT License.
 Copyright (c) 2017 Dorian Stoll (StollD), RockyTV
*/

package adapter

import (
    "github.com/KSP-SpaceDock/SpaceDock-Backend/app"
    "github.com/KSP-SpaceDock/SpaceDock-Backend/middleware"
    "github.com/KSP-SpaceDock/SpaceDock-Backend/objects"
    "github.com/KSP-SpaceDock/SpaceDock-Backend/routes"
    "github.com/KSP-SpaceDock/SpaceDock-Backend/utils"
    "github.com/spf13/cast"
    "gopkg.in/kataras/iris.v6"
)

func init() {
    routes.Register(routes.GET, "/api/adapter/mods/:modid", middleware.Recursion(0), mods_adapter)
}

/*
 Path: /api/adapter/mods/:modid
 Method: GET
 Description: Returns information for one mod
 */
func mods_adapter(ctx *iris.Context) {
    // Get params
    modid := cast.ToUint(ctx.GetString("modid"))

    // Get the mods gameshort
    mod := &objects.Mod{}
    app.Database.Where("id = ?", modid).First(mod)
    if mod.ID != modid {
        utils.WriteJSON(ctx, iris.StatusNotFound, utils.Error("The modid is invalid").Code(2130))
        return
    }
    app.Database.Model(mod).Related(&(mod.Game), "Game")
    if ctx.URLParam("callback") != "" {
        ctx.Redirect("/api/mods/" + mod.Game.Short + "/" + cast.ToString(modid) + "?callback=" + ctx.URLParam("callback"), iris.StatusPermanentRedirect)
        return
    }
    ctx.Redirect("/api/mods/" + mod.Game.Short + "/" + cast.ToString(modid), iris.StatusPermanentRedirect)
}